### ISSUE TITLE
Static timeout values in Conn

### ISSUE THREAD

**Issue by h4cc** [Azolo/websockex#10]
It would be cool, if these two numbers could be part of the Conn-struct and interchangeable via options.

https://github.com/Azolo/websockex/blob/master/lib/websockex/conn.ex#L92
https://github.com/Azolo/websockex/blob/master/lib/websockex/conn.ex#L188

**Comment by Azolo** [comment]
There's no reason I can think of that they couldn't be. Feel like opening a PR?

**Comment by h4cc** [comment]
I am feeling it :dancer:

**Comment by h4cc** [comment]
@Azolo Might you have a look at commit ddc50b4 before PR?

**Comment by Azolo** [comment]
It looks alright, I left a couple of comments. 

But I think I prefer the `Conn` struct flat for the sake of discoverability. Someone can open up `IEx` and do something like:

```elixir
iex> struct(WebSockex.Conn)
%WebSockex.Conn{cacerts: nil, conn_mod: nil, extra_headers: [], host: nil,
 insecure: true, path: nil, port: nil, query: nil, socket: nil, transport: nil,
 socket_connect_timeout: 6000, socket_recv_timeout: 5000}
```

But functionality wise, it looks spot on.

**Comment by h4cc** [comment]
Thanks for your feedback, i updated the implementation.

**Comment by Azolo** [comment]
Closed by #15
